test/unit/nics/bmc_test.rb

```diff
@@ -5,4 +5,39 @@ class BMCTest < ActiveSupport::TestCase
     host = FactoryGirl.build(:host, :managed)
     assert FactoryGirl.build(:nic_bmc, :host => host, :provider => 'ipmi').valid?
   end
+
+  context "no BMC smart proxy exists" do
+    def setup
+      SmartProxy.stubs(:with_features).with('BMC').returns([])
+    end
+
+    test 'requires BMC proxy without subnet' do
+      host = FactoryGirl.build(:host, :managed)
+      bmc_nic = FactoryGirl.build(:nic_bmc, :host => host)
+      host.interfaces << bmc_nic
+      refute host.valid?
+    end
+
+    test 'requires BMC proxy in the same subnet' do
+      host = FactoryGirl.build(:host, :managed)
+      bmc_nic = FactoryGirl.build(:nic_bmc, :with_subnet, :host => host)
+      bmc_nic.subnet.proxies.stubs(:find).returns([])
```

Comment:
I don't see a benefit in stubbing in this case either (also because stubbing AR query methods is more than tricky), if the test builds records using factory girl, it can build exactly those it needs for the test